### PR TITLE
Template tester report

### PR DESCRIPTION
**Pull request by amavinag**
Template tester test report
    
      * Generates the html report if there is any template tested
        within last 24 hours. This can be executed as the downstream
        job for templatest_tester_ jobs in Jenkins.
      * Email notification with this report attached will be configured
        in the Jenkins job.

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit d4ec5e313ef9b7c3a29d76aaf076d3daea9f2f37:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit f41f2931bc6b6d151005285ba40e84cf6b5215d8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
We already include Jinja2 for HTML templates. Do you think you can use the existing base template in `data/` and subclass a new one, then pass through variables and generate HTML from the template?

**Comment by amavinag** [comment]
Thanks Milan, good to know about existing templates. submitted the changes.

**Comment by dajoRH** [comment]
Lint report for commit 397c6171d79de92fd3f22d26d3591f443b8c4456:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 95475948f9538145de0a179c4e657ab0b3f24844:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: scripts/latest_template_tester_report.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    stream_data = get_latest_tested_template_on_stream(api, stream)
+    if stream_data and not get_untested_templates(api, stream_data['group_name']):
+        print "Found tested template for {}".format(stream)
+        print "Gathering tested template data for {}".format(stream)
+        stream_html = [stream_data['template_name'], stream_data['passed_on_providers'],
+                       stream_data['failed_on_providers'], stream_data['group_name'],
+                       stream_data['datestamp']]
+        data = template_env.get_template('template_tester_report.html').render(
+            upstream=stream_html)
+        template_tester = open(log_path.strpath + '/template_tester_results.html', 'w')
```
Comment (mfalesni) [inline scripts/latest_template_tester_report.py]: I'd say add a parameter to the script that will be default to this file to specify the output file. It should use `py.path` functionality to join the path elements:
```python
parser.add_argument("--output", dest="output",
                    help="target file name", default=log_path.join('template_tester_results.html').strpath)
```

then:

```python
with open(filename, 'w') as report:
    report.write(data)
```

Apart from this, just add shebang:
```python
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
```

and chmod +x this file. I tested the functionality, seems working ok.

**Comment by psav** [comment]
Is this extra functionality that the template tester doesn't provide?

**Comment by amavinag** [comment]
psav: yes, this will be a post build script for template tester.

**Comment by dajoRH** [comment]
Lint report for commit 920069cf753fd5d0c849519d36277d9e64bb8121:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**